crates/block-explorers/src/lib.rs

```diff
@@ -536,6 +542,17 @@ mod tests {
         assert!(matches!(resp, ResponseData::Error { .. }));
     }
 
+    #[test]
+    fn can_parse_etherscan_mainnet_invalid_api_key() {
+        let err = json!({
+            "status":"0",
+            "message":"NOTOK",
+            "result":"Missing/Invalid API Key"
+        });
+        let resp: ResponseData<Address> = serde_json::from_value(err).unwrap();
+        assert!(matches!(resp, ResponseData::Error { .. }));
+    }
```

Comment:
this test succeeds on main, so still unclear what's the motivation for this

Reply:
My mistake with writing the test.

It should have been
```rs
 let resp: ResponseData<String> = serde_json::from_value(err).unwrap();
```
or `ResponseData<Option<String>>` like in the contract_abi case I tested on.

I've updated the test but you will see it should fail on main without the updated serializer.

The problem is that the deserializer can match ResponseData of string/option string as a success or an error case. Since success takes precedence it deserializes as a success.

Reply:
Hey @mattsse, just flagging my above comment this is still an issue and the broken case isn't solved.

I see you opted to make your own PR so I'll leave this one unless you would like the change I made in which case I can update it for you or make any updates you might want.